server/command.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+const awsSNSCmd = "awssns"
+
+func (p *Plugin) registerCommands() error {
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          awsSNSCmd,
+		Description:      "Mattermost slash command to interact with AWS SNS",
+		DisplayName:      "AWS SNS",
+		AutoComplete:     true,
+		AutoCompleteHint: "listTopics",
+		AutoCompleteDesc: "List Topics which are subscribed to the channel",
+	}); err != nil {
+		return errors.Wrapf(err, "failed to register awssns command")
+	}
+
+	return nil
+}
+
+// ExecuteCommand Implementing the Command functionality for above commands
+func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	splitCmd := strings.Fields(args.Command)
+	cmd := strings.TrimPrefix(splitCmd[0], "/")
+	action := splitCmd[1]
+
+	if cmd != awsSNSCmd {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         fmt.Sprintf("Unknown Command: " + cmd),
+		}, nil
+	}
+
+	switch action {
+	case "listTopics":
```

Comment:
I'm referencing another plugin here, but would you mind refactoring to use the convention found here? https://github.com/mattermost/mattermost-plugin-agenda/blob/master/server/command.go#L57

Basically, set up each `case` to call a function that handles the logic?  This would really help when adding future slash commands.